File: Practica1/Java/selectionsort.java
import java.util.*;
import java.io.*;
import java.util.Scanner;
import java.io.File;
import java.io.FileNotFoundException;
public class selectionsort {
	public static void swap(int minimo,int i){
		int temp=minimo;
		minimo=i;
		i=temp;
	}
	public static void selectionsort(int[] A){
		int tam= A.length;
		for(int i=0;i<tam-1;i++){
			int minimo=i;
			for(int j=i+1;j<tam;j++){
				if(A[j]<A[minimo]){
					minimo=j;
				}
			}
			int temp=A[minimo];
			A[minimo]=A[i];
			A[i]=temp;
		}
	}
	public static void print(int[] A){
		for(int i:A){
			System.out.print(i+" ");
		}
	}public static void leer(int[] a,int tam) throws FileNotFoundException,IOException{
        int numeroEntero,indice=0;
        File f=new File("../GeneradorAleatorio/Array_"+tam+".txt");
        try(Scanner entrada = new Scanner(f)){
          while (entrada.hasNextInt() && indice<tam) { 
                numeroEntero = entrada.nextInt();
                a[indice]=numeroEntero;
                indice++;
            }
        }catch (FileNotFoundException e) {
        }
	}
	
	public static void main(String args[]) throws FileNotFoundException,IOException
	{
		File file=new File("TiemposJava/Selection_Sort.txt");
        long time1, time2;
        FileWriter fw = new FileWriter(file);
		BufferedWriter bw = new BufferedWriter(fw);
		for(int i=10000;i<=100000;i=i+10000)
        {
          int[] a=new int[i];
          // System.out.println(" indice"+ ( id ));
          leer(a,i);
          selectionsort select = new selectionsort();
          time1=System.currentTimeMillis();
          // time1=System.currentTimeMillis();
          select.selectionsort(a);
          time2=System.currentTimeMillis();
          long tiempo=time2-time1;
          // time2=System.currentTimeMillis();

          System.out.println("Tiempo "+ ( tiempo ) +" s");
          String s=String.valueOf(tiempo);
          
            
            bw.write(s+'\n');
		}
		bw.close();
	} 
}

File: Practica1/C++/selectionsort.cpp
#include <bits/stdc++.h>
#include "../GeneradorAleatorio/generator.h"
using namespace std;
void crear(int *&A,int n)
{
	A=new int[n];
}
void swap(int *xp, int *yp)
{
    int temp = *xp;
    *xp = *yp;
    *yp = temp;
}

void selectionSort(int arr[], int n)
{
    int minimo;
    for (int i = 0; i < n-1; i++){
        minimo = i;
        for (int j = i+1; j < n; j++){
        	if (arr[j] < arr[minimo])
            	minimo = j;
  }
        	swap(&arr[minimo], &arr[i]);


    }
}

void print(int arr[], int tam)
{
    int i;
    for (i=0; i < tam; i++)
        cout << arr[i] << " ";
    cout << endl;
}

int main()
{
    int n=10000;
	double t0;
	double time1;
	int k=0;
	ofstream fs("Tiempos/time_Selection.txt");
	srand(time(NULL));
	for(int i=n;i<=600000;i=i+60000){
		int *A;
		crear(A,i);
		read(A,i,k);
		t0=clock();
		selectionSort(A,i);
		t0=clock()-t0;
		time1= ((double)t0)/CLOCKS_PER_SEC;
		k++;
		fs<<time1<<endl;
	}
    return 0;
}
